### ISSUE TITLE
"You don't have permission to exec" in UI while using Vault OIDC authorization

### ISSUE THREAD

**Issue by beninghton** [hashicorp/nomad#18577]
### Nomad version
Nomad v1.6.0
BuildDate 2023-07-18T18:51:11Z
Revision 87d411f4e3a6a744d328c95708c60c12bae0fb18

### Operating system and Environment details
Fedora release 35 (Thirty Five)

### Issue
We use Vault as OIDC provider and if we authorize via Vault we can't perform allocation exec via UI (both on task/job levels).
![image](https://github.com/hashicorp/nomad/assets/20965960/a2ddfcd6-f4f4-4a86-b89b-01309142709f)
![image](https://github.com/hashicorp/nomad/assets/20965960/d9b84cbb-5d4f-4ff8-a84b-7148dad1c469)

However, If we use the same token that we got from Vault during authorization and use it in Nomad CLI we successfuly perform exec:
```
[root@nomad-server-1 core]# nomad alloc exec -namespace test -i -t 406f41aa-49fc-1bea-6abb-e4b9ba634cd8 /bin/sh
#
```
The policy content:
```
namespace "test" {
  policy = "write"
}
```

Specifying more specific capabilities such as:
`capabilities = ["submit-job","list-jobs","read-job","read-logs","alloc-exec", "read-fs", "alloc-node-exec"]` 
does not help either.

### Reproduction steps
Connect Vault as OIDC provider. 
Assign policy = "write" to namespace the allocation resides.
Login and try to perform Exec.

#### Expected Result
Alloc exec is performed in both Nomad UI and CLI.

#### Actual Result
Alloc exec is performed only in Nomad CLI and denied and UI.

**Comment by the-nando** [comment]
I run into the same issue also without Vault OIDC but with the native Nomad one. The button is disabled in the UI but if enabled by tampering the HTML the exec works fine. As far as I can tell the UI fails to determine if the user has the exec capability on the namespace, possibly related to the use of roles instead of bare policies.

**Comment by philrenaud** [comment]
Hi @beninghton , @the-nando — at first glance, this seems like it is due to the lack of the UI's support for ACL Roles. Good news: the [UI will start supporting them very soon](https://github.com/hashicorp/nomad/pull/18346).

**Comment by philrenaud** [comment]
Marking this as closed — will be resolved with #17770
